tests/pruners_tests/test_median.py

```diff
@@ -116,3 +117,33 @@ def test_median_pruner_n_warmup_steps():
     # A pruner is activated after warm-up steps.
     assert pruner.prune(
         study=study, trial=study._storage.get_trial(trial.trial_id))
+
+
+@pytest.mark.parametrize(
+    'n_warmup_steps,interval_steps,report_steps,expected_prune_steps', [
+        (1, 2, 1, [2, 4]),
+        (0, 3, 10, list(range(1, 30))),
+        (2, 3, 10, list(range(11, 30))),
+        (0, 10, 3, [1, 2, 3, 13, 14, 15, 22, 23, 24]),
+        (2, 10, 3, [4, 5, 6, 13, 14, 15, 25, 26, 27]),
```

Comment:
Please add documentation that tells users to report intermediate values right before calling `Trial.should_prune`?

As can be seen in the test cases, the pruning interval seems to be irregular when `interval_steps` and `report_steps` are relatively prime numbers. I think this is because the pruner sometimes refers to the values of previous steps, and such behaviors are expected. But, users may get confused when they see these irregular intervals.

Reply:
Updated the documentation in `{Percentile,Median}Pruner.__init__`. Could you take another look? I'll improve the docs of `should_prune` in a separate PR.

Reply:
That's great! I think users can clearly understand the pruners' behaviors. Thank you!

Reply:
Rather, thank you for the comments and discussions. It turned out a lot better!

Reply:
> I'll improve the docs of should_prune in a separate PR.
Done in https://github.com/pfnet/optuna/pull/690.